src/v/storage/probe.h

```diff
@@ -34,7 +34,17 @@ class probe {
 
     void corrupted_compaction_index() { ++_corrupted_compaction_index; }
 
-    void segment_created() { ++_log_segments_created; }
+    void segment_created() {
+        ++_log_segments_created;
```

Comment:
why have one counter for created vs removed ? - isn't that active segments? why keep additional state. what are you trying to show here that couldn't be done w/ a simple substraction.